src/Fixer/FunctionNotation/NativeFunctionInvocationFixer.php

```diff
@@ -93,22 +154,91 @@ public function isRisky()
+        }
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    protected function createConfigurationDefinition()
+    {
+        return new FixerConfigurationResolver([
+            (new FixerOptionBuilder('exclude', 'List of functions to ignore.'))
+                ->setAllowedTypes(['array'])
+                ->setAllowedValues([static function (array $value) {
+                    foreach ($value as $functionName) {
+                        if (!\is_string($functionName) || '' === \trim($functionName) || \trim($functionName) !== $functionName) {
+                            throw new InvalidOptionsException(\sprintf(
+                                'Each element must be a non-empty, trimmed string, got "%s" instead.',
+                                \is_object($functionName) ? \get_class($functionName) : \gettype($functionName)
+                            ));
+                        }
+                    }
+
+                    return true;
+                }])
+                ->setDefault([])
+                ->getOption(),
+            (new FixerOptionBuilder('include', 'List of function names or sets to fix. Defined sets are `@internal` (all native functions), `@all` (all global functions) and `@compiler_optimized` (functions that are specially optimized by Zend).'))
+                ->setAllowedTypes(['array'])
+                ->setAllowedValues([static function (array $value) {
+                    foreach ($value as $functionName) {
+                        if (!\is_string($functionName) || '' === \trim($functionName) || \trim($functionName) !== $functionName) {
+                            throw new InvalidOptionsException(\sprintf(
+                                'Each element must be a non-empty, trimmed string, got "%s" instead.',
+                                \is_object($functionName) ? \get_class($functionName) : \gettype($functionName)
+                            ));
+                        }
 
-        $indexes = [];
+                        static $sets = [
+                            self::SET_ALL,
+                            self::SET_INTERNAL,
+                            self::SET_COMPILER_OPTIMIZED,
+                        ];
 
-        for ($index = 0, $count = $tokens->count(); $index < $count; ++$index) {
-            $token = $tokens[$index];
+                        if ('@' === $value[0] && !\in_array($value, $sets, true)) {
+                            throw new InvalidOptionsException(\sprintf('Unknown set "%s", known sets are "%s".', $value, \implode('", "', $sets)));
+                        }
+                    }
 
-            $tokenContent = $token->getContent();
+                    return true;
+                }])
+                ->setDefault([self::SET_INTERNAL])
+                ->getOption(),
+            (new FixerOptionBuilder('scope', 'Only fix function calls that are made within a namespace or fix globally.'))
+                ->setAllowedValues(['global', 'namespaced'])
```

Comment:
I got a little confusion with `global` - there is sth like a `global namespace` in PHP, which is basically code out of any namespace.
Yet here, we are actually fixing everything.

Maybe rename it to sth less confusing, eg `all`, or change option to boolean `fix_only_namespaced: boolean = false` ?

I'm leaving decision up to you here, maybe it misleaded only me ;)

Reply:
I'll it open for a bit of time to get more opinions on it (@julienfalque suggested this https://github.com/FriendsOfPHP/PHP-CS-Fixer/pull/3054#discussion_r139291584 one) :)

Reply:
`fix_only_namespaced` in fact is the same as `ignore_global_space` that Julien suggested ;)

Reply:
I rather don't have a boolean value here. Having a `scope` option of type `string` allows us to extend on later if we want to define other scopes.
Renaming the current possible values `'global', 'namespaced'` is fine by me if someone has a better idea.

Reply:
`'all', 'only_namespaced'` ?

Reply:
> Maybe rename it to sth less confusing, eg `all`

`all` and `namespaced` 👍

Reply:
all done

Reply:
big thanks